lightningd/jsonrpc.c

```diff
@@ -531,6 +531,12 @@ struct json_stream *json_stream_success(struct command *cmd)
 	return r;
 }
 
+struct json_stream *json_stream_success_no_log(struct command *cmd)
+{
+	cmd->jcon->log = NULL;
```

Comment:
Aren't you going to re-enable logging at some point? `json_start` seems like a good candidate to ensure we have a logger in place.

Reply:
I shouldn't use ACK so quickly

Reply:
Does this not suggest that a per-`json_stream` flag is more appropriate?  This may affect other pieces of code.

Reply:
Note that 0.7.1 already has logic to truncate things which would take > 1/64 of the buffer from commit 87cd628f5244b9cb0ca380c06c58722854dfa28c (aka v0.7.0-137-g87cd628f5):

```
	/* Don't immediately fill buffer with giant IOs */
	if (len > log->lr->max_mem / 64) {
		l->skipped++;
		len = log->lr->max_mem / 64;
	}
```

This should fix the crash in this case anyway.

Skipping logging is always kind of dangerous, IMHO, since it leaves blind spots.  If we really want this change, we need to:

1. Print an IO level message telling it that we're not going to log the getlog, eg. "log_io(jcon->log, LOG_IO_OUT, "Suppressing logging", NULL, 0);" and another one
when we reenable.
2. Add `json_stream_log_suppress(const char *msg);` and call it from json_getlog() to disable, and then reenable in json_stream_output_write() where it calls reader_cb (which implies it's sent all the output).

Reply:
emm, compares with#2467 , this commit [3dd4795](https://github.com/ElementsProject/lightning/commit/6cac2c959e41011b73a2452f1345eca0abd7d093) has reduced the number of log entries generated during `getlog io` process

Reply:
> Aren't you going to re-enable logging at some point?

Yes your right, I guess it worked here because the `json_connection` got closed after every command and then initialized (with log enabled) when opened again. But we cannot rely on that.

> Does this not suggest that a per-json_stream flag is more appropriate?

Exactly, then we don't touch `jcon->log` but only this specific [out] stream.